graphix/pattern.py

```diff
@@ -380,7 +381,7 @@ def add_correction_domain(
                 if clifford_gate := c_dict.pop(cmd.node, None):
                     new_cmd = cmd.clifford(clifford_gate)
                 else:
-                    new_cmd = cmd.model_copy()
+                    new_cmd = dataclasses.replace(cmd)
```

Comment:
Using `copy.copy` would be more idiomatic in this context. Additionally, Pydantic’s `model_copy` performed a deep copy, and we need it to handle aliasing correctly.
```suggestion
                    new_cmd = deepcopy(cmd)
```

Below is a test that could be added to `test_pattern.py` to demonstrate the issue:
```python
def test_pattern_aliasing(self) -> None:
    pattern_a = Pattern(input_nodes=[0, 1])
    pattern_b = Pattern(input_nodes=[1])
    pattern_a.add(M(node=0))
    pattern_a.add(X(node=1, domain={0}))
    m = M(node=1)
    pattern_a.add(m)
    pattern_b.add(m)
    pattern_a.standardize()
    pattern_b.simulate_pattern()
```

Reply:
Fixed in f02a260a589aa65f435d236daca476b31cfedc85 .